timemachine/fe/free_energy.py

```diff
@@ -1221,55 +1221,14 @@ def get_flattened_params(initial_state: InitialState) -> NDArray:
         # Add an identity move to the mixture to ensure aperiodicity
         neighbor_pairs = [(StateIdx(0), StateIdx(0))] + neighbor_pairs
 
-    # Setup the barostat to run more often for equilibration
-    equil_barostat_interval = 15
     barostat = context.get_barostat()
-    if barostat is not None:
-        barostat.set_interval(equil_barostat_interval)
 
-    def get_equilibrated_xvb(xvb: CoordsVelBox, state_idx: StateIdx) -> CoordsVelBox:
-        # Set up parameters of the water sampler movers
-        if md_params.water_sampling_params is not None:
-            for mover in context.get_movers():
-                if isinstance(mover, WATER_SAMPLER_MOVERS):
-                    assert water_params_by_state is not None
-                    mover.set_params(water_params_by_state[state_idx])
-        if md_params.n_eq_steps > 0:
-            # Set the movers to 0 to ensure they all equilibrate the same way
-            # Ensure initial mover state is consistent across replicas
-            for mover in context.get_movers():
-                mover.set_step(0)
-
-            params = params_by_state[state_idx]
-            bound_potentials[0].set_params(params)
-
-            context.set_x_t(xvb.coords)
-            context.set_v_t(xvb.velocities)
-            context.set_box(xvb.box)
-
-            xs, boxes = context.multiple_steps(md_params.n_eq_steps, store_x_interval=0)
-            assert len(xs) == 1
-            x0 = xs[0]
-            v0 = context.get_v_t()
-            box0 = boxes[0]
-            xvb = CoordsVelBox(x0, v0, box0)
-        return xvb
-
-    initial_replicas = [
-        get_equilibrated_xvb(CoordsVelBox(s.x0, s.v0, s.box0), StateIdx(i)) for i, s in enumerate(initial_states)
-    ]
-
-    hrex = HREX.from_replicas(initial_replicas)
+    hrex = HREX.from_replicas([CoordsVelBox(s.x0, s.v0, s.box0) for s in initial_states])
```

Comment:
Could revert this as it does corrupt the timings estimate, but since we typically call into `estimate_relative_free_energy_bisection_hrex_impl` which does all the equilibration in the bisection stage it doesn't impact our hot path.

Mostly wanted to avoid having two duplicate implementations of equilibration.

Reply:
> I'm actually not sure why we added equilibration here to begin with. (Do you recall?) 

I am not sure either. Though I vaguely remember `sample` requiring `n_eq_steps > 0` at one point, so maybe to get around that?

>  the other function with a similar interface accepting a list of initial states, run_sims_sequential, also does not do equilibration

To be precise `run_sims_sequential` does do equilibration, if you pass `md_params.n_eq_steps > 0` by calling `sample` (which eventually calls into https://github.com/proteneer/timemachine/blob/master/timemachine/fe/free_energy.py#L601-L613) but does not separate it out.

Reply:
> To be precise run_sims_sequential does do equilibration

Got it, I was confused before. And I was forgetting that the initial states returned from the bisection phase are not equilibrated (only pre-equilibrated, I think?). So equilibration does seem necessary here. The question is whether to have this duplicated code, or do the equilibration as part of the first frame? Is there a third option where we pull out a generic function like `get_equilibrated_xvb` and use it in both places?

Reply:
> And I was forgetting that the initial states returned from the bisection phase are not equilibrated (only pre-equilibrated, I think?)

They are equilibrated as bisection calls `sample` which will perform the equilibration before collecting the samples for bisection, we also update the initial states according: https://github.com/proteneer/timemachine/blob/master/timemachine/fe/rbfe.py#L666-L714. We have incorporated our equilibration behavior into the family of `sample` functions, so think that is right.

Reply:
Ah, OK, I forgot that we do update the bisection initial states with the coordinates from the final frame. So maybe the reason for equilibration here was just for consistency with the other `run_sims_*` functions, even though additional MD equilibration seems unnecessary in our current RBFE implementation.